### ISSUE TITLE
Create a functional interface to journal.js

### ISSUE THREAD

**Issue by jniles** [IMA-WorldHealth/bhima#777]
After #634, the underlying code to many of the journal.js routes was improved (and added a few bugs which @mbayopanda has helped remove).  However, the interface to the journal routes has not improved.  This would be one of the components in #724.

I propose that we refactor the interface to journal.js, creating an HTTP interface (accessible from the client) and a functional interface for other controllers to use.  This interface would look something like this:
```js
journal.post('sale', data)  // returns  promise
.then(function () { console.log('data posted!'); });
```
or
```js
journal.post('distribute_service', data)
.then(function () { console.log('data posted!'); });
```

This would allow our server side controllers to easily do actions like create a cash payment and immediately post it.  While you _can_ currently post to the journal from the server, the interface is really gross.  Here's a sample (from `serviceDist.js`):
```js
function writeToJournal (document_id, userId, details) {
  var deferred = q.defer();
  journal.request('distribution_service', document_id, userId, function (error, result) {
    if (error) {
      return deferred.reject(error);
    }                                                                                                                                
    return deferred.resolve(result);                                                                                                 
  }, undefined, details);                                                                                                            
  return deferred.promise;                                                                                                           
}
```
This is overly complex, requires a callback instead of promises, and is generally difficult to read.  Not sure what priority this is, but as we migrate more validation and creation to the server, we will spend more time writing code that posts to the journal from the server side.
